### PR TITLE
OcXmlLib: Docs reformatting 

### PR DESCRIPTION
**Pull request by PMheart**

### CODE REVIEW

**Review by vit9696 — COMMENTED** [review]
Marked obvious issues.

File: Library/OcXmlLib/OcXmlLib.c [hunk 1]
```diff
@@ -1449,19 +1500,19 @@ PlistDocumentRoot (
...
 
-  return XmlNodeChild(Node, 0);
+  return XmlNodeChild (Node, 0);
 }
 
 XML_NODE *
 PlistNodeCast (
-  XML_NODE         *Node,
-  PLIST_NODE_TYPE  Type
+  IN  CONST XML_NODE   *Node,
```
[the remark this answers is not in the document]
Comment (PMheart) [reply Library/OcXmlLib/OcXmlLib.c]: Should those functions calling `PlistNodeCast ` also have the parameter `Node` changed to `OPTIONAL`?

File: Library/OcXmlLib/OcXmlLib.c [hunk 2]
```diff
@@ -201,21 +218,28 @@ XmlNodeCreate (
...
+
+  @retval  TRUE on successful adding.
+**/
 STATIC
 BOOLEAN
 XmlNodeChildPush (
-  XML_NODE  *Node,
-  XML_NODE  *Child
+  IN OUT  XML_NODE  *Node,
+  IN      XML_NODE  *Child
```
Comment (vit9696) [inline Library/OcXmlLib/OcXmlLib.c:233]: Why is there no ASSERT on NULL child push?

File: Library/OcXmlLib/OcXmlLib.c [hunk 3]
```diff
@@ -272,17 +296,28 @@ XmlNodeChildPush (
...
-  UINT32       ReferenceNumber
+  IN OUT  XML_REFLIST  *References,
+  IN      XML_NODE     *Node,
+  IN      UINT32       ReferenceNumber
   )
 {
   XML_NODE   **NewReferences;
   UINT32     NewRefAllocCount;
 
+  ASSERT (References != NULL);
```
Comment (vit9696) [inline Library/OcXmlLib/OcXmlLib.c:319]: What about Node?

File: Library/OcXmlLib/OcXmlLib.c [hunk 4]
```diff
@@ -318,11 +353,19 @@ XmlPushReference (
...
+  @param[in]  References  A pointer to the XML references.
+  @param[in]  Attributes  XML attributes.
+
+  @return  The real XML node from the one referencing it.
+**/
 STATIC
 XML_NODE *
 XmlNodeReal (
-  XML_REFLIST  *References,
-  CONST CHAR8  *Attributes
```
Comment (vit9696) [inline Library/OcXmlLib/OcXmlLib.c:325]: Both params look optional to me.

File: Library/OcXmlLib/OcXmlLib.c [hunk 5]
```diff
@@ -392,20 +446,28 @@ XmlFreeRefs (
...
-  XML_PARSER         *Parser,
-  XML_PARSER_OFFSET  Offset,
-  CONST CHAR8        *Message
+  IN  CONST XML_PARSER   *Parser,
+  IN  XML_PARSER_OFFSET  Offset,
+  IN  CONST CHAR8        *Message
   )
 {
-  UINT32  Character = 0;
+  UINT32  Character;
```
Comment (vit9696) [inline Library/OcXmlLib/OcXmlLib.c:463]: Missing ASSERTs.

File: Library/OcXmlLib/OcXmlLib.c [hunk 6]
```diff
@@ -447,15 +516,19 @@ XmlParserError (
...
 STATIC
 CHAR8
 XmlParserPeek (
-  XML_PARSER  *Parser,
-  UINT32      N
+  IN  CONST XML_PARSER  *Parser,
+  IN  UINT32            N
   )
 {
   UINT32  Position;
```
Comment (vit9696) [inline Library/OcXmlLib/OcXmlLib.c:534]: No ASSERT?

File: Library/OcXmlLib/OcXmlLib.c [hunk 7]
```diff
@@ -869,22 +981,32 @@ XmlParseContent (
...
+  IN      CONST CHAR8  *Data,
+  IN      UINT32       DataLength
   )
 {
   CHAR8   *NewBuffer;
   UINT32  NewSize;
 
+  ASSERT (Buffer      != NULL);
+  ASSERT (AllocSize   != NULL);
+  ASSERT (CurrentSize != NULL);
```
Comment (vit9696) [inline Library/OcXmlLib/OcXmlLib.c:1008]: What about Data?

File: Library/OcXmlLib/OcXmlLib.c [hunk 8]
```diff
@@ -910,22 +1032,32 @@ XmlBufferAppend (
...
+  IN OUT  CHAR8           **Buffer,
+  IN OUT  UINT32          *AllocSize,
+  IN OUT  UINT32          *CurrentSize,
+  IN      UINT32          Skip
   )
 {
   UINT32  Index;
   UINT32  NameLength;
 
+  ASSERT (Buffer      != NULL);
```
Comment (vit9696) [inline Library/OcXmlLib/OcXmlLib.c:1057]: What about Node?

**Comment by vit9696** [comment]
I see way too many places with ASSERTs missing without any logical sense or OPTIONAL tags. Please fix the ones I spotted and go over the functions to correct the remaining ones as marking them all will be  a waste.

**Pull request merged**